src/middlewares/validateDataWithZod.ts

```diff
@@ -8,9 +8,9 @@ const validateDataWithZod =
 		const result = Schema.safeParse(rawData);
 
 		if (!result.success) {
-			const zodErrors = { errors: result.error.flatten().fieldErrors };
-
-			res.status(422).json(zodErrors);
+			const errors = result.error.flatten().fieldErrors;
+			const message = `${Object.keys(errors).toString()} fields are required!`;
```

Comment:
Looks good so far... But it look like the new zodError format only returns the a string of fields that have errors alongside an "are required" message, even when the error might not necessarily be that the field's information is missing...

May I know the exact intent behind this?

Other than that, every other change seems fine

Reply:
The main intent is to send meaningful message to the client.
This error is more readable than the previous

Reply:
Fair enough then 👍